cdap-metadata-ext-spanner/src/main/java/io/cdap/cdap/metadata/spanner/SpannerMetadataStorage.java

```diff
@@ -25,36 +31,263 @@
+   *     </ul>
+   * </p>
+   * TODO(CDAP-21174): Add the TTl policies for cleanups.
+   */
+  private String metadataTableDdlStatement() {
+    return String.format(
+      "CREATE TABLE IF NOT EXISTS %s (" +
+        "%s STRING(MAX) NOT NULL," +  // metadata_id
+        "%s STRING(MAX) NOT NULL," + // namespace
+        "%s STRING(MAX) NOT NULL," + // entity_type
+        "%s STRING(MAX) NOT NULL," + // name
+        "%s INT64," + // creation-time
+        "%s STRING(MAX)," + // user
+        "%s STRING(MAX)," + // system
+        "%s JSON," + // metadata_column
+        "%s INT64 NOT NULL," +
+        "user_tokens TOKENLIST AS " +
+        "(TOKENIZE_SUBSTRING(USER, ngram_size_min=>1, ngram_size_max=>3)) HIDDEN," +
+        "system_tokens TOKENLIST AS " +
+        "(TOKENIZE_SUBSTRING(SYSTEM, ngram_size_min=>1, ngram_size_max=>3)) HIDDEN," +
+        "text_tokens TOKENLIST AS " +
+        "(TOKENLIST_CONCAT([User_Tokens, System_Tokens])) HIDDEN," +
+        ")PRIMARY KEY (%s) ", // primary key
+      METADATA_TABLE,
+      METADATA_ID_FIELD,
+      NAMESPACE_FIELD,
+      TYPE_FIELD,
+      NAME_FIELD,
+      CREATED_FIELD,
+      USER_FIELD,
+      SYSTEM_FIELD,
+      METADATA_COLUMN_FIELD,
+      VERSION,
+      METADATA_ID_FIELD
+    );
+  }
+
+  /**
+   * <p>Key features of the schema:
+   *     <ul>
+   *         <li>**`metadata_id`:**An Unique which is used to identify the various
+   *               components of the pipeline metadata.
+   *         </li>
+   *         <li>**`namespace`:** It contains the namespace associated with a given entity.
+   *         </li>
+   *         <li>**`entity_type`:** It contains the type of entity.
+   *         </li>
+   *         <li>**`props_name`:** Contains the name of the properties/tags associated to an
+   *             entries in metadata table.
+   *         </li>
+   *         <li>**`props_scope`:** Contains the scope of the properties/tags associated to an
+   *             entries in metadata table.
+   *         </li>
+   *         <li>**`props_value`:** Contains the value of the properties/tags associated to an
+   *             entries in metadata table.
+   *         <li>value_tokens is the Tokenlist of Props_Value Column on which we create search
+   *         indexes and use of key:value type searches.
+   *         respectively.</li>
+   *     </ul>
+   * </p>
+   */
+  private String metadataPropsTableDdlStatement() {
+    return String.format(
+      "CREATE TABLE IF NOT EXISTS %s (" +
+        "%s STRING(MAX) NOT NULL," +  // metadata_id
+        "%s STRING(MAX) NOT NULL," + // namespace
+        "%s STRING(MAX) NOT NULL," + // entity_type
+        "%s STRING(MAX) NOT NULL," + // name
+        "%s STRING(MAX)," + // scope
+        "%s STRING(MAX)," + // value
+        "value_tokens TOKENLIST AS " +
+        "(TOKENIZE_SUBSTRING(Props_Value, ngram_size_min=>1, ngram_size_max=>3)) HIDDEN," +
+        ")PRIMARY KEY (%s, %s, %s) ," + // primary key
```

Comment:
`// primary key` comment is obvious as it is next to the code: `")PRIMARY KEY (%s) "`

It would be useful if the primary key column name is mentioned.